client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -108,14 +108,20 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 			this.props.trackEvent( 'Clicked Save Settings Button' );
 		};
 
-		submitForm = () => {
+		submitForm = ( callback ) => {
 			const { fields, settingsFields, siteId, jetpackSettingsUISupported } = this.props;
 			this.props.removeNotice( 'site-settings-save' );
 
-			this.props.saveSiteSettings( siteId, pick( fields, settingsFields.site ) );
+			const settingsRequests = [
+				this.props.saveSiteSettings( siteId, pick( fields, settingsFields.site ) )
+			];
+
 			if ( jetpackSettingsUISupported ) {
-				this.props.updateSettings( siteId, pick( fields, settingsFields.jetpack ) );
+				settingsRequests.push( this.props.updateSettings( siteId, pick( fields, settingsFields.jetpack ) ) );
 			}
+
+			Promise.all( settingsRequests )
```

Comment:
@oskosk actually, this will require a major refactor of `lib/track-form` - since as you mentioned, callbacks are used for `setState()`, but this way we lose the opportunity to return the promise (see `handleAutosavingRadio` for example). Maybe we could leave that for another PR, what do you think?